src/App.svelte

```diff
@@ -0,0 +1,216 @@
+<script>
+  import { db } from "./firebase.js";
+  import { onMount } from "svelte";
+  import AddItemForm from "./AddItemForm.svelte";
+  import GroceryItem from "./components/GroceryItem.svelte";
+
+  let name;
+  let newItemList = {};
+
+  let groceryList = [];
+  let shareIconImage = "./images/share-icon.svg";
+  let bellIconImage = "./images/bell-regular.svg";
+  let redoIconImage = "./images/redo-solid.svg";
+  let noteIconImage = "./images/sticky-note-regular.svg";
+  let calendarIconImage = "./images/calendar.svg";
+  let dateString;
+  let isEditing = false;
+  let xxx = "";
+
+  let now = new Date(),
+    month,
+    day,
+    year;
+
+  onMount(() => {
+    db.collection("grocery").onSnapshot((snapData) => {
+      groceryList = snapData.docs;
+    });
+    (month = "" + (now.getMonth() + 1)),
+      (day = "" + now.getDate()),
+      (year = now.getFullYear());
+    if (month.length < 2) month = "0" + month;
+    if (day.length < 2) day = "0" + day;
+    dateString = [year, month, day].join("-");
+  });
+
+  function removeItem(itemId) {
+    console.log("The event works");
+    console.log(itemId);
+    db.collection("grocery").doc(itemId).delete();
+  }
+
+  // async function updateItem(itemId, newItemValue) {
+  //    const response = await db.collection("grocery").doc(itemId).update({ name: newItemList[itemId] });
+  //   newItemList[itemId] = '';
+  //   isEditing = false;
+
+  // }
+
+  function editItem() {
+    isEditing = true;
+  }
+
+  function updateItem(itemId, inputEl) {
+    console.log(itemId);
+    console.log(inputEl);
+    console.log(newItemList);
+    db.collection("grocery").doc(itemId).update({ name: inputEl });
+    console.log("no update");
+    isEditing = false;
+
+    // console.log(isEditing);
+    // isEditing = false;
+  }
+</script>
+
+<header />
+<main>
+  <h1>Grocery list</h1>
+  <h3>
+    {dateString} <img src={calendarIconImage} alt="{calendarIconImage} image" />
+  </h3>
+  <AddItemForm />
+
+  <ul>
+    {#each groceryList as listItem}
+      <GroceryItem
+        item={listItem.data()}
+        {removeItem}
```

Comment:
I must confess I don't know this sugar syntax to add events! Where did you see it to use like this instead of `removeItem={removeItem}`?

Reply:
Did you struggled to use follow the tutorial to [passing props as events](https://svelte.dev/tutorial/component-events)?

Reply:
@sandrina-p No I didn't really understand the logic.